src/java/org/apache/cassandra/db/streaming/CassandraCompressedStreamReader.java

```diff
@@ -89,11 +89,15 @@ public SSTableMultiWriter read(DataInputPlus inputPlus) throws Throwable
                 cis.position(section.lowerPosition);
                 in.reset(0);
 
+                long lastBytesRead = 0;
                 while (in.getBytesRead() < sectionLength)
                 {
                     writePartition(deserializer, writer);
                     // when compressed, report total bytes of compressed chunks read since remoteFile.size is the sum of chunks transferred
-                    session.progress(filename + '-' + fileSeqNum, ProgressInfo.Direction.IN, cis.chunkBytesRead(), totalSize);
+                    long bytesRead = cis.chunkBytesRead();
+                    long bytesDelta = bytesRead - lastBytesRead;
+                    lastBytesRead = bytesRead;
+                    session.progress(filename + '-' + fileSeqNum, ProgressInfo.Direction.IN, bytesRead, bytesDelta, totalSize);
```

Comment:
Could pull `filename + '-' + fileSeqNum` out of the while loop too rather than hope the compiler is smart enough. Looks true for most of the other call points too.